src/NServiceBus.Core/Recoverability/FirstLevelRetries/FirstLevelRetriesBehavior.cs

```diff
@@ -42,7 +40,6 @@ class FirstLevelRetriesBehavior : Behavior<ITransportReceiveContext>
                 {
                     storage.ClearFailuresForMessage(pipelineUniqueMessageId);
                     context.Message.Headers[Headers.FLRetries] = numberOfFailures.ToString();
-                    notifications.Errors.InvokeMessageHasFailedAFirstLevelRetryAttempt(numberOfFailures, context.Message, ex);
```

Comment:
not sure if we really should remove the notification here.
We still invoke "Message Has Failed a First Level Retry Attempt" when the message fails on the first processing attempt which isn't a retry attempt but we no longer raise the same event when it actually has failed a FLR attempt and will be forwarded to SLR? Seems to be as wrong as previously (while the number of invocations now matches the expected number, but on the wrong processing attempts).